RealmSwift/Tests/RealmTests.swift

```diff
@@ -31,80 +31,79 @@ class RealmTests: TestCase {
             }
         }
     }
 
     func testPath() {
-        XCTAssertEqual(Realm.defaultPath, Realm().path)
+        XCTAssertEqual(Realm.defaultPath, try! Realm().path)
         XCTAssertEqual(testRealmPath(), realmWithTestPath().path)
     }
 
     func testReadOnly() {
         autoreleasepool {
-            XCTAssertEqual(Realm().readOnly, false)
+            XCTAssertEqual(try! Realm().readOnly, false)
         }
-        let readOnlyRealm = Realm(path: Realm.defaultPath, readOnly: true, error: nil)!
+        let readOnlyRealm = try! Realm(path: Realm.defaultPath, readOnly: true)
         XCTAssertEqual(true, readOnlyRealm.readOnly)
         XCTAssertEqual(3, readOnlyRealm.objects(SwiftIntObject).count)
 
-        assertThrows(Realm(), "Realm has different readOnly settings")
+        assertThrows(try! Realm(), "Realm has different readOnly settings")
     }
 
     func testSchema() {
-        let schema = Realm().schema
+        let schema = try! Realm().schema
         XCTAssert(schema as AnyObject is Schema)
         XCTAssertEqual(1, schema.objectSchema.filter({ $0.className == "SwiftStringObject" }).count)
     }
 
     func testDefaultPath() {
-        let defaultPath =  Realm().path
+        let defaultPath =  try! Realm().path
         XCTAssertEqual(Realm.defaultPath, defaultPath)
 
         let newPath = defaultPath.stringByAppendingPathExtension("new")!
         Realm.defaultPath = newPath
         XCTAssertEqual(Realm.defaultPath, newPath)
-        XCTAssertEqual(Realm().path, Realm.defaultPath)
+        XCTAssertEqual(try! Realm().path, Realm.defaultPath)
     }
 
     func testInit() {
-        XCTAssertEqual(Realm().path, Realm.defaultPath)
-        XCTAssertEqual(Realm(path: testRealmPath()).path, testRealmPath())
-        assertThrows(Realm(path: ""))
+        XCTAssertEqual(try! Realm().path, Realm.defaultPath)
+        XCTAssertEqual(try! Realm(path: testRealmPath()).path, testRealmPath())
+        assertThrows(try! Realm(path: ""))
     }
 
     func testInitFailable() {
-        var error: NSError?
         autoreleasepool {
-            Realm(path: Realm.defaultPath, readOnly: false)
-            XCTAssertNil(error)
+            try! Realm(path: Realm.defaultPath, readOnly: false)
         }
 
         NSFileManager.defaultManager().createFileAtPath(Realm.defaultPath,
             contents:"a".dataUsingEncoding(NSUTF8StringEncoding, allowLossyConversion: false),
             attributes: nil)
-        XCTAssertNil(Realm(path: Realm.defaultPath, readOnly: false, error: &error), "Should not throw with error")
-        XCTAssertNotNil(error)
+        do {
+            try Realm(path: Realm.defaultPath, readOnly: false)
+            XCTFail("Realm creation should have failed")
+        } catch {
+        }
 
-        assertThrows(Realm(path: Realm.defaultPath, readOnly: false, error: nil))
-        assertThrows(Realm(path: Realm.defaultPath, readOnly: false))
```

Comment:
The scenario being tested there does not have an equivalent with swift exceptions.